zerver/actions/message_send.py

```diff
@@ -430,30 +433,50 @@ def followed_topic_notification_recipients(setting: str) -> Set[int]:
 
     def get_ids_for(f: Callable[[ActiveUserDict], bool]) -> Set[int]:
         """Only includes users on the explicit message to line"""
-        return {row["id"] for row in rows if f(row)} & message_to_user_id_set
+        return {row.id for row in rows if f(row)} & message_to_user_id_set
 
     active_user_ids = get_ids_for(lambda r: True)
     online_push_user_ids = get_ids_for(
-        lambda r: r["enable_online_push_notifications"],
+        lambda r: r.enable_online_push_notifications,
     )
 
     # We deal with only the users who have disabled this setting, since that
     # will usually be much smaller a set than those who have enabled it (which
     # is the default)
     dm_mention_email_disabled_user_ids = get_ids_for(
-        lambda r: not r["enable_offline_email_notifications"]
+        lambda r: not r.enable_offline_email_notifications
     )
     dm_mention_push_disabled_user_ids = get_ids_for(
-        lambda r: not r["enable_offline_push_notifications"]
+        lambda r: not r.enable_offline_push_notifications
     )
 
     # Service bots don't get UserMessage rows.
     um_eligible_user_ids = get_ids_for(
-        lambda r: not r["is_bot"] or r["bot_type"] not in UserProfile.SERVICE_BOT_TYPES,
+        lambda r: not r.is_bot or r.bot_type not in UserProfile.SERVICE_BOT_TYPES,
     )
 
+    if stream and stream.is_support_stream and not message_sender.is_bot:
+        realm = get_realm_by_id(realm_id)
+        authorized_user_ids = [row.id for row in rows if
+                                 check_access_based_on_stream_topic_view_policy(row, stream)]
+
+        topic_creator_user_id = get_topic_creator_user_id(
+            realm_id,
+            stream.recipient.id,
+            stream_topic.topic_name
+        )
+
+        possible_um_eligible_user_ids = set(
+            authorized_user_ids 
+            + ([topic_creator_user_id] if topic_creator_user_id is not None else []) 
+            + [sender_id]
+        )
```

Comment:
This might be cleaner with just a `set.add(sender_id)` type line after initializing the starting set.

Reply:
Done.